server/chat-plugins/mafia.ts

```diff
@@ -3774,6 +3782,62 @@ export const commands: Chat.ChatCommands = {
 			this.parse('/mafia hostbanhelp');
 		},
 
+		gamebanname: 'namehostban',
+		namegameban: 'namehostban',
+		hostbanname: 'namehostban',
+		namehostban(target, room, user, connection, cmd) {
+			if (!target) return this.parse('/help mafia namehostban');
+			room = this.requireRoom();
+			this.checkCan('warn', null, room);
+
+			const [targetUser, rest] = this.splitOne(target);
+			const [string1, string2] = this.splitOne(rest);
+			let duration, reason;
+			if (parseInt(string1)) {
+				duration = parseInt(string1);
+				reason = string2;
+			} else {
+				duration = parseInt(string2);
+				reason = string1;
+			}
+
+			if (!duration) duration = 2;
+			if (!reason) reason = '';
+			if (reason.length > 300) {
+				throw new Chat.ErrorMessage("The reason is too long. It cannot exceed 300 characters.");
+			}
+
+			const userid = toID(targetUser);
+			const commandType = cmd.includes('hostban') ? 'hostban' : 'gameban';
+			if (Punishments.hasRoomPunishType(room, userid, `MAFIA${commandType.toUpperCase()}`)) {
+				throw new Chat.ErrorMessage(`User '${targetUser}' is already ${commandType}ned in this room.`);
+			} else if (Punishments.hasRoomPunishType(room, userid, `MAFIAGAMEBAN`)) {
+				throw new Chat.ErrorMessage(`User '${targetUser}' is already gamebanned in this room, which also means they can't host.`);
+			} else if (Punishments.hasRoomPunishType(room, userid, `MAFIAHOSTBAN`)) {
+				user.sendTo(room, `User '${targetUser}' is already hostbanned in this room, but they will now be gamebanned.`);
+				this.parse(`/mafia unhostban ${targetUser}`);
```

Comment:
Is there an issue with a user being host and game banned at the same time? If so, you might want to handle the reversal differently than issuing a command which may fail/will add to modlog that they were unhostbanned.

Reply:
I'll change this as I think it'd be better if people could be host and game banned at the same time, but this wasn't as much a conscience decision right now as much as just keeping things as they were.